### PR TITLE
Run CI on push to any branch and schedule all workflows hourly

### PR DESCRIPTION
**Pull request by mvorisek**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | no

#### Summary

- triggering on push to any branch is important to allow testing before PR is made
- test default branch hourly is important to provided actual build state for the default branch

### CODE REVIEW

File: .github/workflows/coding-standards.yml [hunk 1]
```diff
@@ -7,8 +7,9 @@ on:
       - "master"
```
Comment (mvorisek) [inline .github/workflows/coding-standards.yml:7]: I even belive, we should trigger CI for every PR as we might want to made PR into another PR

**Comment by greg0ire** [comment]
> triggering on push to any branch is important to allow testing before PR is made

People can and should test locally, then they can make a draft PR IMO

**Comment by mvorisek** [comment]
> 
> 
> > triggering on push to any branch is important to allow testing before PR is made
> 
> People can and should test locally, then they can make a draft PR IMO

yes, this is what it does - if you are in other repo, CI results are logged there, not here...

**Comment by mvorisek** [comment]
@greg0ire this is also what Symfony does.

doctrine/orm does trigger on PR only (not on push) - but I will submit there a PR shortly ;-)

File: .github/workflows/coding-standards.yml (hunk 1 above)
Comment (mvorisek) [reply .github/workflows/coding-standards.yml]: done

**Comment by greg0ire** [comment]
> locally

I mean on their computer, not on their fork.

**Comment by mvorisek** [comment]
I know ;-), but why not in GH which is the standard for almost every other project now :)

File: .github/workflows/coding-standards.yml [hunk 2]
```diff
@@ -1,14 +1,10 @@
-name: "Coding Standards"
+name: "CS"
```
Comment (greg0ire) [inline .github/workflows/coding-standards.yml:1]: Already discussed in another PR, that's a no for me.

File: .github/workflows/coding-standards.yml [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
 
 on:
   pull_request:
-    branches:
-      - "*.x"
-      - "master"
   push:
-    branches:
-      - "*.x"
-      - "master"
```
Comment (greg0ire) [inline .github/workflows/coding-standards.yml:11]: How can we having double the jobs when making a PR from origin to origin?

Comment (mvorisek) [reply .github/workflows/coding-standards.yml:11]: that is impossible as far as I know

File: .github/workflows/coding-standards.yml (hunk 2 above)
Comment (mvorisek) [reply .github/workflows/coding-standards.yml]: I do not prefer it from UX perspective, but ok with me if you do not want to change your decision here :)

File: .github/workflows/coding-standards.yml (hunk 3 above)
Comment (greg0ire) [reply .github/workflows/coding-standards.yml:11]: > normally

Yes, normally… keyword… sometimes some maintainers will push to origin, deliberately or not.

Comment (mvorisek) [reply .github/workflows/coding-standards.yml:11]: if "not normally", then I am aware GH displays both results in PR checks, but that is fine, it does not affect the code in any way, it is purerly how GH display results 8-)

Comment (greg0ire) [reply .github/workflows/coding-standards.yml:11]: I don't know, @morozov what do you think?

Comment (mvorisek) [reply .github/workflows/coding-standards.yml:11]: that is IMO correct, as once with your direct (source branch) commit and once with (simulated) merge commit (into target/PR branch)

Comment (morozov) [reply .github/workflows/coding-standards.yml:11]: > It's not just display IMO, the tests are really run twice.

That's the point. This is the reason why I wanted to introduce these filters.

Comment (mvorisek) [reply .github/workflows/coding-standards.yml:11]: I know, but what is the real issue with it?

When I was implementing GH actions for other repos, I noticed this too, but I think that it is fully fine when CI tests are run everywhere, also, here is my research on top php repos:
- https://github.com/symfony/symfony/blob/5.x/.github/workflows/tests.yml
- https://github.com/sebastianbergmann/phpunit/blob/master/.github/workflows/ci.yml
- https://github.com/composer/composer/blob/master/.github/workflows/continuous-integration.yml
- https://github.com/laravel/framework/blob/8.x/.github/workflows/tests.yml

all of them trigger CI on push event

Comment (mvorisek) [reply .github/workflows/coding-standards.yml:11]: ok to merge?

**Comment by morozov** [comment]
Closing as I don't see a value in applying this change.

**Comment by mvorisek** [comment]
@morozov I am very disappointed as allowing CI to run on push in very and industry standard practice across broad spectrum of open-source projects. Please see https://github.com/doctrine/dbal/pull/4349#discussion_r506721571

**Comment by mvorisek** [comment]
> People can and should test locally, then they can make a draft PR IMO

Please explain **SHOULD**, draft PR is definitelly not a good solution for anything that should not be merged

**Comment by mvorisek** [comment]
> People can and should test locally, then they can make a draft PR IMO

@greg0ire please explain **SHOULD**, draft PR is definitelly not a good solution for anything that should not be merged

**Comment by greg0ire** [comment]
I mean people should attempt to run the test they can run locally before opening a PR.

**Comment by mvorisek** [comment]
> 
> 
> I mean people should attempt to run the test they can run locally before opening a PR.

Yes! Many of them just create a branch, push, and want to see the CI result. This is perfect for quick and small fixes when local setup migh be very time consuming even for one DB.

So this PR seems perfectly legit to me.

**Comment by greg0ire** [comment]
> Many of them just create a branch, push, and want to see the CI result.

Isn't that what draft PRs are for?

**Comment by mvorisek** [comment]
> draft PR is definitelly not a good solution for anything that should not be merged

obviously no :)

**Comment by greg0ire** [comment]
Says who? You?

**Comment by mvorisek** [comment]
Yes, me! - like verify something etc. No need to create a draft PR and "polite" the history

**Comment by greg0ire** [comment]
You mean pollute.

I don't think there are that many people just creating branches and never opening a PR, that seems unlikely. Besides, draft PRs are great in case you want to get early feedback and avoid wasting time on something that will never get merged. Of course, this works better when you start with documentation and explanations and finish with code rather than doing the opposite.

**Comment by mvorisek** [comment]
I am saying, that there a cases, when PR is not needed. I think, we can agree, that every change/experiment does not need a push to PR nor needs an attention from others.

To be objective, I have provided links to popular repos - in https://github.com/doctrine/dbal/pull/4349#discussion_r506721571 - that this is "very common practice".

**Comment by greg0ire** [comment]
> I think, we can agree, that every change/experiment does not need a push to PR nor needs an attention from others.

We agree on that. We can also agree that it's a waste to have double the jobs when a maintainer pushes their branch to the origin repository.

> To be objective, I have provided links to popular repos - in #4349 (comment) - that this is "very common practice".

That's a bandwagon fallacy. Let's ignore that.

**Comment by mvorisek** [comment]
> maintainer pushes their branch to the origin repository

this should really not be an argument as thanks to GH, GH minutes are cheap/free for open source

**Comment by greg0ire** [comment]
> what objective criteria do you honor then?

I'm not sure, but I try to ignore fallacies.

> this should really not be an argument as thanks to GH, GH minutes are cheap/free for open source

It is the argument made here nonetheless : https://github.com/doctrine/dbal/pull/4349#discussion_r506718451 discuss that with its author.

**Comment by greg0ire** [comment]
Anyway, this PR should have been made against https://github.com/doctrine/.github

**Comment by mvorisek** [comment]
@morozov please review the discussing with @greg0ire here. I find this PR very helpfull as any push runs a CI test which is very helpfull information for developer

**Comment by mvorisek** [comment]
> 
> 
> Anyway, this PR should have been made against https://github.com/doctrine/.github

thanks, done, here https://github.com/doctrine/.github/pull/11

**Comment by morozov** [comment]
I'm locking this conversation as too heated and not bringing any value to the project. @mvorisek if you don't stop pushing your agenda, harassing maintainers, and creating an unnecessary sense of urgency, you will be banned from this project.

**Pull request closed without merging**